cmake/Intel.cmake

```diff
@@ -18,29 +18,33 @@ if(REPRO)
     add_definitions(-DREPRO)
     set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -qno-opt-dynamic-align")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -qno-opt-dynamic-align")
+elseif(DEBUG)
+    add_definitions(-DDEBUG)
```

Comment:
CMAKE_BUILD_TYPE is DEBUG.
Use appropriate `CMAKE_<lang>_FLAGS_DEBUG`

Reply:
Please submit a follow-up PR to clean up the entire cmake build system. I am only making minimal changes to an existing cmake build system that already has DEBUG and REPRO defined. I don't think we want to make these changes now and rerun all the regression tests, but I may be mistaken.

Reply:
If I make the changes as you suggested, I'll break the existing builds. You'll see why in a minute when I walk you throuh the legacy code.